scrapy/core/engine.py

```diff
@@ -400,6 +400,18 @@ def _spider_idle(self) -> None:
             assert isinstance(ex, CloseSpider)  # typing
             self.close_spider(self.spider, reason=ex.reason)
 
+    def close_spider_before_start(self, spider: Spider, reason: str = "cancelled"):
+        if self.slot is not None:
+            raise RuntimeError("Engine slot is already assigned. Use self.close_spider")
+
+        self.start()
```

Comment:
This doesn't look correct because start() returns a Deferred, but also is it OK to call it here at all if we are closing anyway?

Reply:
@wRAR 
I didn't place it at beginning. But to common `close_spider` method has several checks for engine being started. Not sure we need to have use it though, just think that the code inside it is useful. 

I've tried to avoid it, but neither of approaches I've considered allowed to get the proper result. The only other option I see so far is to simply set close reason to spider and leave it in piece.

How do you think to do it better?

Reply:
In any case if you call it you need to wait until it completes.